pyvista/plotting/plotting.py

```diff
@@ -278,6 +278,22 @@ def __init__(
         if self.theme.antialiasing:
             self.enable_anti_aliasing()
 
+        # widgets
+        self.camera_widgets = []
+        self.box_widgets = []
+        self.box_clipped_meshes = []
+        self.plane_widgets = []
+        self.plane_clipped_meshes = []
+        self.plane_sliced_meshes = []
+        self.line_widgets = []
+        self.slider_widgets = []
+        self.threshold_meshes = []
+        self.isovalue_meshes = []
+        self.spline_widgets = []
+        self.spline_sliced_meshes = []
+        self.sphere_widgets = []
+        self.button_widgets = []
```

Comment:
Ideally we should move all these into `WidgetHelper.__init__`. But for that we would have to take care to call `super().__init__` in our own `__init__`s.

Here's the method resolution order for `Plotter`:
```py
>>> pv.Plotter.mro()
[<class 'pyvista.plotting.plotting.Plotter'>, <class 'pyvista.plotting.plotting.BasePlotter'>, <class 'pyvista.plotting.picking.PickingHelper'>, <class 'pyvista.plotting.widgets.WidgetHelper'>, <class 'object'>]
```
We'd definitely have to call `super().__init__()` in `BasePlotter.__init__()`, but that might be enough since `PickingHelper` has no `__init__()` of its own. And the only reason for this would be purity; not sure it's worth it.

Then again it might be good form to prepare `BasePlotter` for cooperative multiple inheritance if it's easy to do. Because if someone does
```py
class DuperPlotter(BasePlotter, DuperType):
    ...
```
then `BasePlotter.__init__()` would be called first as per the MRO, and `DuperType.__init__()` wouldn't have a chance to run unless we call `super().__init__` in `BasePlotter`.


_However_, one thing that could be tricky is deciding what args to pass on to `super().__init__()`. Because we might want to pass on args not caught by us, but we also must prepare for the possibility that `super()` resolves to `object` which doesn't take any parameters in its `__init__()`. It might work to add `*args, **kwargs` to our `__init__()` signatures and pass those on... when everything Just Works then we'd be passing on empty `args` and `kwargs` to `object.__init__()`.

Reply:
Looks like we've been using class attributes to avoid multiple inheritance, and to fix this I've done as you've suggested; moved the applicable to the `__init__`s of the subclasses. Not easy to wrap my head around, but learned a bit about [MRO](https://www.educative.io/edpresso/what-is-mro-in-python).

For anyone who wants to read up on multiple cooperative inheritance:
https://stackoverflow.com/a/27134600

Reply:
There's also Raymond Hettinger's [Python's `super()` considered super!](https://rhettinger.wordpress.com/2011/05/26/super-considered-super/) as a great resource.